File: Programming101/week0/python_simple_problem_set/sum_of_divisors.py
def sum_of_divisors(n):
    sum_ = 0
    for number in range(1, n+1):
        if n % number == 0:
            sum_ += number
    return sum_


def main():
    print(sum_of_divisors(8))
    print(sum_of_divisors(7))
    print(sum_of_divisors(1))
    print(sum_of_divisors(1000))

if __name__ == '__main__':
    main()

File: FrontendJavaScript/week0/2-A-Set-Of-Problems/set1/sumOfDivisors.js
'use strict';

var sumOfDivisors = function(n){
  var sum = 0;
  var i = n;
  for(i; i>0; i--) {
    if (n % i === 0) {
      sum += i;
    }
  }
  return sum;
};

console.log(sumOfDivisors(8) === 15);
console.log(sumOfDivisors(7) === 8);
console.log(sumOfDivisors(1) === 1);
console.log(sumOfDivisors(1000) === 2340);
